python/anvill/binja.py

```diff
@@ -323,14 +337,82 @@ def visit(self, program, is_definition, add_refs_as_defs):
 
         mem = program.memory()
 
-        ref_eas = set()
+        ref_eas: Set[int] = set()
         ea = self._bn_func.start
         max_ea = self._bn_func.highest_address
         self._fill_bytes(program._bv, mem, ea, max_ea, ref_eas)
 
+        # Collect typed register info for this function
+        for block in self._bn_func.llil:
+            for inst in block:
+                register_information = self._extract_types(program._bv, inst.operands, inst)
+                for reg_info in register_information:
+                    loc = Location()
+                    loc.set_register(reg_info[0].upper())
+                    loc.set_type(reg_info[1])
+                    if reg_info[2] is not None:
+                        # fill_bytes misses some references, catch what we can
+                        ref_eas.add(reg_info[2])
+                        # print(f"inst_addr: {hex(inst.address)}, reg_info[2]: {reg_info[2]}, ref_eas: {ref_eas}")
+                        # assert reg_info[2] in ref_eas
+                        # assert reg_info 1 is a pointer, then reg2 should be in ea
+                        loc.set_value(reg_info[2])
+                    loc.set_address(inst.address)
+                    self._register_info.append(loc)
+        # ea = effective_address
+        # there is a distinction between declaration and definition
+        # if the function is a declaration, then Anvill only needs to know its symbols and prototypes
+        # if its a definition, then Anvill will perform analysis of the function and produce information for the func
         for ref_ea in ref_eas:
             program.try_add_referenced_entity(ref_ea, add_refs_as_defs)
 
+    def _extract_types_mlil(self, bv, item_or_list, initial_inst: mlinst) -> List[Tuple[str, Type, Optional[int]]]:
+        """
+        This function decomposes a list of MLIL instructions and variables into a list of tuples
+        that associate registers with pointer information if it exists.
+        """
+        results = []
+        if isinstance(item_or_list, list):
+            for item in item_or_list:
+                results += self._extract_types_mlil(bv, item, initial_inst)
```

Comment:
use `rests.extend`, or pass `results` into `self._extract...` and change it to something like `def _accumulate_types_mlil`